---
title: "workflow"
author: "Yupei You"
date: "`r format(Sys.time(), 'create date: 6/5 2019 last updated: %d-%B-%Y  %H:%m')`"
---
<style>
.column-left{
  float: left;
  width: 50%;
  text-align: left;
}
.column-center{
  display: inline-block;
  width: 100%;
  text-align: center;
}
.column-right{
  float: right;
  width: 50%;
  text-align: left;
}
</style>

### DTW using based on scrappie model (version1 with SquiggleKit)

Do this by using SquiggleKit MotifSeq.py[^3]:

**Input:**
scrappie model
signal.tsv

**output:**
best local match of the motif model in the singal

[^3]: https://github.com/Psy-Fer/SquiggleKit

#### Transform data to suit the SquiggleKit input

SquiggleKit require the signal data to be presented as .tsv file with following format:
```{bash eval = F}
GXB01170_20180329_FAH60584_GA50000_sequencing_run_LLAAB035083_26949_read_15463_ch_199_strand.fast5	12611	all	589	564	554	566	540	548	570	566	596	660	683	588	601	629	639	610	607	594	599	543	573	596	593	624	613	614	637	606	618	617	548	537	564	549	565	556	514	541	537	535	559	528	573	555	546	627	552	547	558	569	553	577	621	615	627	600	607	586	562	435	439	457	448	462	462	494	581	577	536	557	529	571	597	555	540	550	520	552	541	526	531	545	536	592	614	614	605	597	610	594	599	612	615	597	586	592	583	592	582	588	588	579	604	579	596	595	600	594	590	589	584	582	584	573	582	577	605	639	641	653	651	762	817	807	824	842	842	805	826	827	815	810	808	819	816	809	818	823	793	778	787	801	781	801	806	822	823	802	807	808	801	794	818	775	761	745	734	714	721	704	686	693	671	649	630	628	645	686	677	701	634	617	617	545	562	548	553	553	552	
```

I selected the region in the previous [table: tombo output](#boundary_region)

The raw data start from **3694(1586+2108) to 3945(1817+20+2108)**:
```{bash eval = F}
cd /home/youyupei/PhD_proj/cDNA_dataset_Analyses/tiny_set/quried_read

SCRIPT_PATH=/home/youyupei/PhD_proj/Python/
fast5_file=/home/youyupei/PhD_proj/cDNA_dataset_Analyses/tiny_set/quried_read/queried_read_false/fast5s/DESKTOP_4SQHSSK_20180117_FAH31144_MN18871_sequencing_run_SHSY5Y_cDNA_bcode_test_35833_read_48338_ch_43_strand.fast5


python $SCRIPT_PATH/fast5_to_tsv.py $fast5_file boundary_signal.tsv 3694 3945
```

Before applying the DTW, I plotted out the expected signal as well as the observed signal:

```{bash eval = F}
cd /opt/shared/PhD_project/Nanopore-Splicing-Site/example
python3 ../visulisation/plot_scrappie_model.py motif_true.model model_plot

python /visulisation/plot_squiggle.py DESKTOP_4SQHSSK_20180117_FAH31144_MN18871_sequencing_run_SHSY5Y_cDNA_bcode_test_35833_read_48338_ch_43_strand.fast5 exon_junc_20_bases.png 3694 3945

```
<center>

**in plot_squiggle.py the input has been updated, instead of the start and end point, the new script require a <position on the transcript> and a <window size>**

![Raw data](../docs/img/exon_junc_20_bases.png){width=75%}


![Expected squiggle](../docs/img/model_plot_ture.png){width=75%}



![False model](../docs/img/model_plot_motif_false.png){width=75%}



</center>





#### DTW using SquiggleKit: MotifSeq.py



In SquiggleKit, DTW is applied using [mlpy.dtw_subsequence](http://mlpy.sourceforge.net/docs/3.5/dtw.html#subsequence-dtw) in [mlpy](http://mlpy.sourceforge.net/docs/3.5/) package:

  mlpy.dtw_subsequence:

  * **It's a local alignment**
  * Use **Manhattan distance**
  
  MotifSeq.py:

  * Input: .tsv file, signal start from the 4th element in each row
  * dtw_subsequence in MotifSeq.py compares the raw signal with the scrappie model (Doesn't use the variance).
  * z-score normalization is applied to the queried read's signal.

Input of MotifSeq.py is saved in `Nanopore-Splicing-Site/example`:

Bash command (ran locally):
```{bash eval=F}
cd /opt/shared/PhD_project/Nanopore-Splicing-Site/example


echo "false model:"
python $shared/SquiggleKit/MotifSeq.py -s boundary_signal.tsv -m motif_false.model
echo "true model:"
python $shared/SquiggleKit/MotifSeq.py -s boundary_signal.tsv -m motif_true.model

```
Here is the result
```{bash eval=F}
false model:

/home/youyupei/PhD_proj/cDNA_dataset_Analyses/tiny_set/quried_read/queried_read_false/fast5s/DESKTOP_4SQHSSK_20180117_FAH31144_MN18871_sequencing_run_SHSY5Y_cDNA_bcode_test_35833_read_48338_ch_43_strand.fast5 Dist: 48.871340492238865 pos: 50 , 166 Dist from Start 50 Length: 116


true model:

/home/youyupei/PhD_proj/cDNA_dataset_Analyses/tiny_set/quried_read/queried_read_false/fast5s/DESKTOP_4SQHSSK_20180117_FAH31144_MN18871_sequencing_run_SHSY5Y_cDNA_bcode_test_35833_read_48338_ch_43_strand.fast5 Dist: 58.51271588303034 pos: 33 , 233 Dist from Start 33 Length: 200
```
   








