<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Yupei You" />


<title>correct_prodiction_f15_t4_t_likelihood</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">workflowr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li>
  <a href="license.html">
    <span class="fa fa-id-card-o"></span>
     
    License
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/youyupei/workflowr">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
<li>
  <a href="To_Do_List.html">
    <span class="fa fa-check-square-o"></span>
     
    TOdo list
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">correct_prodiction_f15_t4_t_likelihood</h1>
<h4 class="author">Yupei You</h4>
<h4 class="date">Created: 23 Sep 2019, last updated: 23-September-2019 00:09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-09-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>workflowr/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190306code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190306)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190306code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190306)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyouyupeiworkflowrtree79e4ce4993ca51c1e0e1dca5b625803053432413targetblank79e4ce4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/youyupei/workflowr/tree/79e4ce4993ca51c1e0e1dca5b625803053432413" target="_blank">79e4ce4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyouyupeiworkflowrtree79e4ce4993ca51c1e0e1dca5b625803053432413targetblank79e4ce4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  analysis/Meeting/
    Untracked:  analysis/Reading/
    Untracked:  analysis/Software/
    Untracked:  analysis/TALON.Rmd
    Untracked:  analysis/Workflow/
    Untracked:  analysis/f15_t4_t_likelihood.Rmd

Unstaged changes:
    Deleted:    analysis/10_Apr_Meeting_Summary.Rmd
    Deleted:    analysis/18_Ma_ with_HJ_and_MK.Rmd
    Deleted:    analysis/5_Mar_with_HJ_and_MK_.Rmd
    Deleted:    analysis/FLAIR.Rmd
    Deleted:    analysis/Journal_club.Rmd
    Deleted:    analysis/Matthew_2016.Rmd
    Deleted:    analysis/SquiggleKit.Rmd
    Deleted:    analysis/Squiggle_alignment.Rmd
    Deleted:    analysis/Tombo resquiggle.Rmd
    Deleted:    analysis/Using sequins with RNA sequencing.Rmd
    Deleted:    analysis/cDNA_R1_102_1.Rmd
    Deleted:    analysis/cDNA_data_preprocessing.Rmd
    Deleted:    analysis/cDNA_workflow.Rmd
    Deleted:    analysis/candidate_generator.Rmd
    Deleted:    analysis/f15_t4_mad_norm.Rmd
    Deleted:    analysis/fetch_single_from_tombo.Rmd
    Deleted:    analysis/get_representative_squiggle.Rmd
    Deleted:    analysis/inaccurate_transcript_checking.Rmd
    Deleted:    analysis/self_developed_DTW.Rmd
    Deleted:    analysis/snakemake_pipline.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>I selected <strong>32</strong> transcripts with depth between 30 ~ 300<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>. In this page, I used t distribution with degree of freedom = 1 to calculate the distance between observed data and simulated squiggles.</p>
<pre class="r"><code>library(ggplot2)
library(reshape2)
datapath = &quot;/home/ubuntu/PhD_proj/pipeline/Validation/&quot;
tested_tranID = read.table(paste0(datapath,&quot;tested_transID.txt&quot;),header = F)[,1] 
print(tested_tranID)</code></pre>
<pre><code> [1] R1_102_1 R1_13_1  R1_22_2  R1_31_2  R1_33_1  R1_33_2  R1_41_1 
 [8] R1_41_2  R1_71_2  R1_91_1  R1_92_2  R2_116_3 R2_117_1 R2_117_3
[15] R2_140_1 R2_150_1 R2_26_1  R2_26_2  R2_28_1  R2_28_2  R2_28_3 
[22] R2_32_1  R2_32_2  R2_38_1  R2_38_2  R2_41_1  R2_45_4  R2_59_1 
[29] R2_59_2  R2_6_2   R2_7_1   R2_72_1 
32 Levels: R1_102_1 R1_13_1 R1_22_2 R1_31_2 R1_33_1 R1_33_2 ... R2_72_1</code></pre>
</div>
<div id="per-site-accuracy" class="section level1">
<h1>per site accuracy</h1>
<p>Firstly, the performance has been evaluated per splicing site.</p>
<p>The overall accruacy across tested splicing sites (splicing sites with only one candidate have been filtered out):</p>
<pre class="r"><code>prediction = prediction[which(prediction$num_of_candidates &gt; 1),]

prediction$accuracy_manhattan = prediction$manhattan_correct/prediction$mapped_reads
prediction$accuracy_z_score = prediction$z_score_correct/prediction$mapped_reads
prediction$accuracy_loglik_score = prediction$log_correct/prediction$mapped_reads

prediction$accuracy_mapper_support   = prediction$num_of_support/prediction$mapped_reads
############temp!!!! caused by incorrect count of mapper supports####
prediction$accuracy_mapper_support[prediction$accuracy_mapper_support &gt;1] = 1


#print(paste0(&quot;Manhattan accuracy: &quot;, mean(prediction$accuracy_manhattan)))
#print(paste0(&quot;z_score accuracy: &quot;, mean(prediction$accuracy_z_score)))
print(paste0(&quot;log likelihood (t distribution, v = 1) accuracy: &quot;, mean(prediction$accuracy_loglik_score)))</code></pre>
<pre><code>[1] &quot;log likelihood (t distribution, v = 1) accuracy: 0.627236406578748&quot;</code></pre>
<pre class="r"><code>print(paste0(&quot;mapper accuracy: &quot;, mean(prediction$accuracy_mapper_support)))</code></pre>
<pre><code>[1] &quot;mapper accuracy: 0.851714277016799&quot;</code></pre>
<p><strong>The overall accuracy per site is not bad, however it is still lower than the minimap2. This is acceptable because minimap2 pushes the reads aligning to the canonical sites nearby, which may also introduce some indels around the splicing site. Our method could hopefully find better support if a minor splicing site exists. Using log likelihood (under normal assumption) seems the best metrics to calculate the distance.</strong></p>
<div id="accuracy-density-plot" class="section level3">
<h3>Accuracy Density plot</h3>
<pre class="r"><code>data &lt;- melt(prediction[c(&quot;accuracy_loglik_score&quot;,&quot;accuracy_mapper_support&quot;)])</code></pre>
<pre><code>No id variables; using all as measure variables</code></pre>
<pre class="r"><code>ggplot(data,aes(x=value, fill=variable)) + geom_density(alpha=0.25) + ggtitle(&quot;Accuracy density plot&quot;) +xlab(&quot;Accuracy per splicing site&quot;)  + theme(plot.title = element_text(hjust = 0.5, size = 15))</code></pre>
<p><img src="figure/f15_t4_t_likelihood.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>########histogram + density##############
#data &lt;- melt(prediction[c(&quot;accuracy_loglik_score&quot;,&quot;accuracy_manhattan&quot;, &quot;accuracy_z_score&quot;, &quot;accuracy_mapper_support&quot;)])#&quot;accuracy_manhattan&quot;, &quot;accuracy_z_score&quot;,
#ggplot(data,aes(x=value, fill=variable)) + geom_histogram(alpha = 0.25,position=&quot;dodge&quot;)   +  theme(plot.title = element_text(hjust = 0.5)) + geom_density(alpha=0.25) # + ggtitle(&quot;Accuracy Density &quot;)</code></pre>
<p><strong>Accuracy Density plot:</strong> the densities using different distance metrics do not differ greatly from each other. The most interesting part is that we could observe a minor peak for the mapper accuracy, which might be interesting cases for out method.</p>
</div>
<div id="boxplot-dtw-vs-minimap2-in-terms-of-splicing-site-selection." class="section level3">
<h3>Boxplot, DTW vs Minimap2 in terms of splicing site selection.</h3>
<pre class="r"><code>data &lt;- melt(data.frame(
  #manhattan = (prediction$manhattan_correct-prediction$num_of_support)/prediction$mapped_reads,
  #zscore = (prediction$z_score_correct-prediction$num_of_support)/prediction$mapped_reads,
  log_likelihood = (prediction$log_correct-prediction$num_of_support)/prediction$mapped_reads))

ggplot(data,aes(y=value, x=variable, fill = variable)) +  
  geom_boxplot(alpha=0.4) + labs(y = &#39;&#39;,x = &#39;&#39;) +
  ggtitle(&quot;DTW vs minimap2 in splicing site characterisition&quot;) +
  xlab(&quot;Distance metrics&quot;)  + 
  ylab(&quot;Percentage&quot;) + 
  theme(plot.title = element_text(hjust = 0.5, size = 15)) + 
  scale_y_continuous(labels=scales::percent)</code></pre>
<p><img src="figure/f15_t4_t_likelihood.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /> <strong>DTW vs minimap2 in splicing site characterisition: </strong> In this boxplot, the y axis is calculated by <span class="math display">\[ \frac{\#\ of\ correctly \ assign\ squiggles\ by\ DTW - \#\ of\ reads\ correctly\ mapped\ to\ the\ same\ splicing\ site\ by\ minimap2}{\#\ of\ reads\ mapped\ to\ the\ splicing\ site}\]</span>. Positive value means out method correctly assigns more junction reads than minimap2. The plot shows that they are very similar, some outliers are observed <strong>and need further investigation.</strong></p>
</div>
</div>
<div id="per-squiggle-accuracy" class="section level1">
<h1>Per squiggle accuracy</h1>
<p>In this section, the performance is evaluated over each squiggle assignment.</p>
<pre class="r"><code>library(ggplot2)
library(reshape2)

#load file
ncol = 50
#score_f15_t4 = read.csv(file = &quot;~/PhD_proj/pipeline/Validation/score_f15_t4_merge.csv&quot;,header = F,col.names = seq(1, ncol))

score_f15_t4 = read.csv(file = &quot;~/PhD_proj/pipeline/Validation/f15_t4_t_likelihood_merge.csv&quot;,header = F,col.names = seq(1, ncol))
while(sum(is.numeric(score_f15_t4$X1)) &gt; 0 || sum(is.na(score_f15_t4$X1))) {
  ncol = ncol + 10
  score_f15_t4 = read.csv(file = &quot;~/PhD_proj/pipeline/Validation/f15_t4_t_likelihood_merge.csv&quot;,header = F,col.names = seq(1, ncol))
}


# fast5 filenames
  filenames = score_f15_t4$X1
  #junctionID = c(1)
  #for (i in 8900:length(filenames)){
  #  if (filenames[i] == filenames[i-1]){
  #    junctionID[i] = junctionID[i-1] + 1
  #  }else{
  #    junctionID[i] = 1
  #  }
  #}

# likelihood_score (single candidate junctions removed)
likelihood_score = cbind(filenames, score_f15_t4[,seq(5,ncol,3)])
likelihood_score = likelihood_score[!is.na(likelihood_score$X5),]
has_multi_candidates = apply(likelihood_score[,-1],1,function(x) sum(!is.na(x)) &gt; 1)
likelihood_score = likelihood_score[has_multi_candidates,]

# distance of best matched candidate
min_dist = apply(likelihood_score[,-1],1,function(x) min(x, na.rm=T))
# bool that the corresponding prodiction is correct or not
correctness = apply(likelihood_score[,-1],1,which.min) == 1</code></pre>
<p>The overall accruacy across tested squiggles (squiggles with only only candidates have been filtered out):</p>
<pre class="r"><code>print(sum(correctness)/length(correctness))</code></pre>
<pre><code>[1] 0.6335404</code></pre>
<div id="best-dtw-distance-distribution" class="section level3">
<h3>Best DTW distance distribution</h3>
<pre class="r"><code>data &lt;- data.frame(likelihood_score,min_dist = min_dist, correctness = correctness)

# distribution of min dist (density plot)
ggplot(data,aes(x=min_dist, fill=correctness)) + 
  scale_fill_manual(labels = c(&quot;False positives&quot;, &quot;True Positives&quot;), values = c(&quot;blue&quot;, &quot;red&quot;))  +
  guides(fill=guide_legend(title=NULL,reverse = T)) +
  geom_histogram(aes(y=3*..count../sum(..count..)),alpha = 0.7,position=&quot;dodge&quot;) + 
  geom_density(alpha=0.3) +
  ggtitle(&quot;Distribution of best match distance&quot;) +
  xlab(&quot;Smallest distance metrics&quot;)  + 
  ylab(&quot;Density&quot;) + 
  scale_y_continuous(sec.axis = sec_axis(~./3, name = &quot;Persentage (histgram)&quot;)) + 
  xlim(1,3) +
  theme(plot.title = element_text(hjust = 0.5, size = 15))</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/f15_t4_t_likelihood.Rmd/distribution%20plot-1.png" width="672" style="display: block; margin: auto;" /> <strong>Plot Distribution of best match distance:</strong> This plot shows the distribution of the best (smallest) distance across all the candidates for each junction squiggle.The squiggles are grouped by whether or not it is correctly assigned to the correct candidate.</p>
<pre class="r"><code>true_reads = likelihood_score[correctness == T,]
false_reads = likelihood_score[correctness == F,]

true_reads_dist = apply(true_reads[,-1],1,function(x) min(x, na.rm = T))
false_reads_dist = apply(false_reads[,-1],1,function(x) min(x, na.rm = T))


false_diff = apply(false_reads[,-1],1,function(x) abs(min(x, na.rm=T) - x[1]))
#false_diff = apply(false_reads[,-1],1,function(x) abs(min(x, na.rm=T) - x[order(x)[2]]))
true_diff = apply(true_reads[,-1],1,function(x) abs(x[1] - x[order(x)[2]]))

ggplot() + 
  geom_density(aes(x = false_diff,fill = &quot;false positive&quot;), alpha = 0.3)  + 
  geom_density(aes(x = true_diff, fill = &quot;true positive&quot;), alpha = 0.3) + 
  scale_color_manual(&quot;Legend&quot;,values = c(&quot;red&quot;,&quot;blue&quot;),aesthetics = &quot;fill&quot;) +
  xlab(&quot;Difference of distance&quot;) +
  ggtitle(&quot;Distribution of difference between true candidate \n and best aligned false candidate&quot;) +
  theme(plot.title = element_text(hjust = 0.5, size = 15))</code></pre>
<p><img src="figure/f15_t4_t_likelihood.Rmd/difference%20density%20(true%20vs%20best%20false)-1.png" width="672" style="display: block; margin: auto;" /> <strong>For the squiggles that assigned to false candidates, the difference between the chosen one and the true one is relatively smaller, which is expected.</strong></p>
</div>
<div id="d-density-plot" class="section level3">
<h3>2d density plot:</h3>
<pre class="r"><code>ggplot() + 
  geom_point(aes(x=true_reads_dist,y=true_diff,color = &quot;true positive&quot;),size = 0.01,alpha = 0.2) + 
  geom_point(aes(x=false_reads_dist,y=false_diff,color = &quot;false positive&quot;),size = 0.01,alpha = 0.8) + 
  geom_density_2d(aes(x=true_reads_dist,y=true_diff,fill = &quot;true positive&quot;,color = &quot;true positive&quot;)) + 
  geom_density_2d(aes(x=false_reads_dist,y=false_diff,fill = &quot;false positive&quot;,color = &quot;false positive&quot;)) + 
  xlab(&quot;hardness of matching (distance)&quot;)+ 
  ylab(&quot;confidence (difference of distance)&quot;)+
  scale_color_manual(&quot;Legend&quot;,values = c(&quot;blue&quot;,&quot;red&quot;)) + 
  ylim(0,1) + 
  xlim(1,3)</code></pre>
<p><img src="figure/f15_t4_t_likelihood.Rmd/2d%20density%20(true%20vs%20best%20false)-1.png" width="672" style="display: block; margin: auto;" /> <strong>y-axis</strong>: match score difference (true vs best false) <strong>x-axis</strong>: score of best matched candidate</p>
<pre class="r"><code>true_reads = likelihood_score[correctness == T,]
false_reads = likelihood_score[correctness == F,]

true_reads_dist = apply(true_reads[,-1],1,function(x) min(x, na.rm = T))
false_reads_dist = apply(false_reads[,-1],1,function(x) min(x, na.rm = T))


#false_diff = apply(false_reads[,-1],1,function(x) abs(min(x, na.rm=T) - x[1]))
false_diff = apply(false_reads[,-1],1,function(x) abs(min(x, na.rm=T) - x[order(x)[2]]))
true_diff = apply(true_reads[,-1],1,function(x) abs(x[1] - x[order(x)[2]]))

ggplot() + 
  geom_density(aes(x = false_diff,fill = &quot;false positive&quot;), alpha = 0.3)  + 
  geom_density(aes(x = true_diff, fill = &quot;true positive&quot;), alpha = 0.3) + 
  scale_color_manual(&quot;Legend&quot;,values = c(&quot;red&quot;,&quot;blue&quot;),aesthetics = &quot;fill&quot;) +
  xlab(&quot;Difference of distance&quot;) +
  ggtitle(&quot;Distribution of difference between best aligned candidate \n and second best aligned false candidate&quot;) +
  theme(plot.title = element_text(hjust = 0.5, size = 15))</code></pre>
<p><img src="figure/f15_t4_t_likelihood.Rmd/difference%20density%20(best%20vs%20second%20best%20false)-1.png" width="672" style="display: block; margin: auto;" /> <strong>In real data analysis, alignment score of true candidate is unknown. But we can calculate the difference between the best and second best score. Compare this plot to the previous one, the density peak for the false postives are much higher, which means the true candidate can not always get the second best score in false positives. the true positive score density should remain the some to the previous one.</strong></p>
</div>
<div id="d-density-plot-1" class="section level3">
<h3>2d density plot:</h3>
<pre class="r"><code>ggplot() + 
  geom_point(aes(x=true_reads_dist,y=true_diff,color = &quot;true positive&quot;),size = 0.01,alpha = 0.2) + 
  geom_point(aes(x=false_reads_dist,y=false_diff,color = &quot;false positive&quot;),size = 0.01,alpha = 0.8) + 
  geom_density_2d(aes(x=true_reads_dist,y=true_diff,fill = &quot;true positive&quot;,color = &quot;true positive&quot;)) + 
  geom_density_2d(aes(x=false_reads_dist,y=false_diff,fill = &quot;false positive&quot;,color = &quot;false positive&quot;)) + 
  xlab(&quot;hardness of matching (distance)&quot;)+ 
  ylab(&quot;confidence (difference of distance)&quot;)+
  scale_color_manual(&quot;Legend&quot;,values = c(&quot;blue&quot;,&quot;red&quot;)) + 
  ylim(0,1) + 
  xlim(1,3)</code></pre>
<p><img src="figure/f15_t4_t_likelihood.Rmd/2d%20density%20(best%20vs%20second%20best%20false)-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>y-axis</strong>: match score difference <span class="math display">\[|best\ score - second\ best\ score|\]</span></p>
<p><strong>x-axis</strong>: score of best matched candidate</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.4 (2018-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.3 LTS

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reshape2_1.4.3 ggplot2_3.2.1 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       knitr_1.23       magrittr_1.5     workflowr_1.4.0 
 [5] MASS_7.3-49      munsell_0.5.0    colorspace_1.4-1 rlang_0.4.0     
 [9] plyr_1.8.4       stringr_1.4.0    tools_3.4.4      grid_3.4.4      
[13] gtable_0.3.0     xfun_0.7         withr_2.1.2      git2r_0.25.2    
[17] htmltools_0.3.6  yaml_2.2.0       lazyeval_0.2.2   rprojroot_1.3-2 
[21] digest_0.6.19    tibble_2.1.3     crayon_1.3.4     fs_1.3.1        
[25] glue_1.3.1       evaluate_0.14    rmarkdown_1.13   labeling_0.3    
[29] stringi_1.4.3    compiler_3.4.4   pillar_1.4.2     scales_1.0.0    
[33] backports_1.1.4  pkgconfig_2.0.2 </code></pre>
</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>The upperbound was set in order to reduce the run time. There are totally 34 transcripts have 30 ~ 300 mapped reads, I removed R1_92_1 and R2_60_2 from the list because it runs too slowly (large number of reads and exon junctions)<a href="#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
