---
title: "Ideas about the direction at the start of the project"
author: "yupei"
date: "2019-03-18"
output: workflowr::wflow_html
---

## 1.Introduction

After talked with Heejung and Mike, there are a couple ideas that worth to be tried.
The project is going to follow the steps listed below:

1. Determine the candidate spicing sites and motifs.
  + We will firstly focus on the canonical splicing site. Because most of the splicing happen at those sites.
  + For all the combinations of potential splicing sites in concern, list all the motifs that could be the joints between exons.
2. **Generate representitive squiggle for each of the motifs.**
3. Squiggle matching. (DTW)

## 2. Generate representitive squiggle for each of the motifs

The main discussion of the meeting were on how to generate a representitive squiggle for a given genomic sequence. Some ideas have come out and worth to be tried:

#### 2.1 Method 1: Generate the squiggle directly from the pore model.

Oxford Nanopore Technology (ONT) provides pore models for their sequencer, which were trained by the company. The model describes the expected cureent signal for each k-mer with its variance. `Scrappie squiggle` is a possible program developed by ONT and does the squiggle simulation. 

**Problems: Currently, it seems that there isn't any RNA model. The available options are "raw_r94", "rgrgr_r94", "rgrgr_r941", "rgrgr_r10", "rnnrf_r94".**

####2.2 Method 2: Get Model from spink-ins

Spike-in provides ground truth of the underlying genomic sequence. Hopefully, the spike-ins contain the motifs in interest. The idea is to find the corresponding signal data to the candidate motifs. 

#####2.2.1
There are two methods found to match/align the events of squiggle to the ref genome: `nanopolish eventalign` and `tombo resquiggle`. Both of them treat the raw data as a series of events.

#####2.2.2
1. Use the method in 2.1, we can simulate squiggles for given motifs
  + However, the simulated squiggle may bia to the model provided by ONT. 

2. Match the simulated squiggle to the spike-ins:
  + We know the exactly sequences of spike-ins so the exactly number of occurances of each k-mer.
  + We know the exon in spike-ins, so that we can narrow down the searching area to the splicing sites.
  
3. Use the matched squiggles from spike-ins to construct a representative squiggle for a motif in interest.


