---
title: "workflow"
author: "Yupei You"
date: "`r format(Sys.time(), 'last updated: %d-%B-%Y  %H:%m')`"
---

## To Do

* Enable resquiggle without transcript reference (Tombo script)
* Coding the new restriction of DTW path
* Talk with Heejung about all probabilities go to 0/1
* Deep Sequins data analysis
* Match short and long read analysis
* Write up the paper except for the real RNA anaylsis part.
  

## To Do (relatively low priority)
* Investigate the frequency of the **current spike** (Terry mentioned *robust statistics*).

* Check drift: Statistics of current level to check whether or not the average level will vary in differet channel or different time in a same channel

* Test the speed variation in a single read.

* Generate presentitave model using spike-in data.

* Check the accuracy of squiggle assignment given by Tombo (Maybe by squencing the regions near splicing sites)


## Done

* Think about penalty for shrink in DTW

cDNA data analysis:

  * Select Spike-in reads from cDNA dataset
  * Pair each fastq read with its raw signal.
  * Successfully run tombo resquiggle
  * (11/4/2019)Generate random motif in different lengths and check their occurrances within spike-in sequence.
  * (23/4/2019)Prepare a slides for Terry's group meeting 
  * Figure out setting in minimap2
  * able to extract signal from .fast5 file
  * able to extract squiggle from Tombo resquiggle output
  * Learn about the scappie squiggle (10/5/2019)
  * Figure out how does FLAIR work. 
  * Check the k-mer distribution in the condidates (is there any k-mer bias to junction/squins)
  * Investigate some specific cases in the following situations:
  
    1. Check signal alignment of site in which a false candidate get very good score and much better than the true one.
    2. The site with very low accuraccy when running dtw
    3. The site with large discrepency of the support from dtw or minimap2
    4. **Tips:** Ignore confusing junctions for which we are not sure the exact transcript the reads come from.
  
  * Play with the spike-in reads: 

	1. Separate spike-in reads into training set and validation set (validation set comes from the different isoform with the training set).
	2. For validation set, randomly pick some exon boundary and infer the potential motifs from mapped spike-in reads.
	3. Check the number of occurrences of the motifs in the training set, tune the length of motifs to guarantee enough number of occurrences. 
	4. Clip out squiggles from motifs as well as the squiggles from corresponding exon boundary in validation set by using Tombo.
	5. Try to use DTW to match the squiggles 
	  
	  * use subsequence_dtw in python package mlpy
	  * use mean and dwell time from scrappie squiggle model as input

* Modify the [DTW](self_developed_DTW.html) I used so that it can take into account the variance.

* Test single site R1_102_1

* Snakemake the [pipeline](snakemake_pipline.html).


